### DISCUSSION TITLE
Support for rqlite clusterable db

### DISCUSSION THREAD

**Discussion by phi0x** [diesel-rs/diesel#2698] [Ideas]
I'm trying to use the bitwarden_rs project for my password manager. I wanted to ensure that my password manager is using a db which is lightweight and redundant/High Availability capable.

diesel-rs supports sqlite3, which rqlite is based on. Would love for diesel-rs to add support for rqlite so I could cluster my db that my bitwarden_rs password manager service utilizes.

I'm running IoT based hardware and also some small cloud vm's (256-512mb ram) and rqlite/sqlite3 are perfect for this type of implementation. Bitwarden_rs utilizes diesel-rs and thus they will not take on extra DB implementations as they are fully integrated with diesel-rs and thus they noted I should either seek to see if rqlite can work with diesel-rs or if diesel-rs can implement compatibility.

Would love to have this! thanks!

**Comment by weiznich** [comment]
We do not have any plans to add support for another database system to diesel itself, especially if non of the maintainers is using that database system
That written: Diesel is written in such a way that it is possible to add support for unsupported database systems outside of diesel as third party crate and to implement custom connections for already supported database systems. See the diesel-oci and the diesel firebird backend implementation for existing examples.

**Comment by phi0x** [comment]
Ok thank you!
